From d335b4da50b79dc03517d3189c55ca0b299b12ad Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 8 Sep 2016 13:59:27 +0300
Subject: [PATCH 0857/1345] pp3: mtu change abort if pool resizing failed

commit  20b3a9b5ea6900d16094536b4f27203aab9b845e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

MTU change reallocates pool with new size buffer.
Abort the changing and exit with error if reallocation failed

Change-Id: I27b0f694158bdd06276746b3c12505e64fbe2a48
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37488
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../net/ethernet/marvell/pp3/net_dev/mv_netdev.c   |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/pp3/net_dev/mv_netdev.c b/drivers/net/ethernet/marvell/pp3/net_dev/mv_netdev.c
index 03fcf13..d592d34 100644
--- a/drivers/net/ethernet/marvell/pp3/net_dev/mv_netdev.c
+++ b/drivers/net/ethernet/marvell/pp3/net_dev/mv_netdev.c
@@ -3027,8 +3027,9 @@ static int mv_pp3_change_mtu_internals(struct net_device *dev, int mtu)
 	dev->mtu = mtu;
 
 	if (long_pool) {
-		mv_pp3_pool_long_sw_init(long_pool, long_pool->headroom,
-						MV_RX_PKT_SIZE(mtu));
+		if (mv_pp3_pool_long_sw_init(long_pool, long_pool->headroom,
+					     MV_RX_PKT_SIZE(mtu)))
+			return -1;
 
 		if (pp3_fw_bm_pool_set(long_pool) < 0)
 			pr_warn("%s: FW long pool update failed\n", dev->name);
-- 
1.7.9.5

